### ISSUE TITLE
Segmentation fault (core dumped) in Python 3

### ISSUE THREAD

**Issue by mchburch** [mfranberg/libplinkio#21]
Hi,

I am trying to do the following in Python:
'plink_obj = WritablePlinkFile(file_prefix,list_of_Samples)'

Then, I get the following error:
'Segmentation fault (core dumped)'

However, when I convert the file down to Python version 2.7 everything works fine. I have the same version of plinkio on both Python 2.7 and 3.5 (plinkio 0.9.7) so I am not quite sure what is happening here. Any help would be appreciated!

Thank you,

Myson

**Comment by brianzhang01** [comment]
I am seeing the same behavior: a segmentation fault (core dumped) in Python 3 when trying to create a WritablePlinkFile. [Using gdb](https://stackoverflow.com/a/2664232), I was able to get a stacktrace, of which the few calls nearest the seg fault are:

Thread 1 "python" received signal SIGSEGV, Segmentation fault.
0x00007ffff7ef41f5 in __strlen_avx2 () from /lib64/libc.so.6
Missing separate debuginfos, use: dnf debuginfo-install gsl-2.4-8.fc30.x86_64
(gdb) bt
#0  0x00007ffff7ef41f5 in __strlen_avx2 () from /lib64/libc.so.6
#1  0x00007ffff7e226c3 in strdup () from /lib64/libc.so.6
#2  0x00007fffd59ba5bf in fam_create (fam_file=fam_file@entry=0x7fffffffcf30, path=path@entry=0x555557557040 "temp.fam",
    samples=samples@entry=0x5555571fe6d0, num_samples=num_samples@entry=300) at src/fam.c:114
#3  0x00007fffd59b9ac2 in pio_create (plink_file=plink_file@entry=0x7fffffffcf20, plink_file_prefix=<optimized out>,
    samples=samples@entry=0x5555571fe6d0, num_samples=300) at src/plinkio.c:105
#4  0x00007fffd59be196 in plinkio_create (self=<optimized out>, args=<optimized out>) at py-plinkio/cplinkio.c:343

I am using Python 3.7.3 and plinkio==0.9.7.

**Comment by leecbaker** [comment]
Same thing, but on mac:
```
Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_platform.dylib      	0x00007fff6bc79e52 _platform_strlen + 18
1   libsystem_c.dylib             	0x00007fff6bb29cd8 strdup + 18
2   cplinkio.cpython-37m-darwin.so	0x000000010bd3910c fam_create + 284
3   cplinkio.cpython-37m-darwin.so	0x000000010bd38abf pio_create + 255
4   cplinkio.cpython-37m-darwin.so	0x000000010bd3be35 plinkio_create + 261
5   org.python.python             	0x0000000106530dd3 _PyMethodDef_RawFastCallKeywords + 235
6   org.python.python             	0x000000010653045c _PyCFunction_FastCallKeywords + 41
7   org.python.python             	0x00000001065c5a6c call_function + 628
8   org.python.python             	0x00000001065bea47 _PyEval_EvalFrameDefault + 6767
9   org.python.python             	0x0000000106530830 function_code_fastcall + 106
10  org.python.python             	0x00000001065311a3 _PyObject_Call_Prepend + 131
11  org.python.python             	0x000000010656e632 slot_tp_init + 80
12  org.python.python             	0x000000010656b35e type_call + 172
13  org.python.python             	0x00000001065302d2 _PyObject_FastCallKeywords + 358
14  org.python.python             	0x00000001065c5ad2 call_function + 730
15  org.python.python             	0x00000001065bea47 _PyEval_EvalFrameDefault + 6767
16  org.python.python             	0x00000001065c62cd _PyEval_EvalCodeWithName + 1698
17  org.python.python             	0x0000000106530424 _PyFunction_FastCallKeywords + 212
18  org.python.python             	0x00000001065c5ad9 call_function + 737
19  org.python.python             	0x00000001065beae2 _PyEval_EvalFrameDefault + 6922
20  org.python.python             	0x0000000106530830 function_code_fastcall + 106
21  org.python.python             	0x00000001065c5ad9 call_function + 737
22  org.python.python             	0x00000001065beae2 _PyEval_EvalFrameDefault + 6922
23  org.python.python             	0x0000000106530830 function_code_fastcall + 106
24  org.python.python             	0x00000001065c5ad9 call_function + 737
25  org.python.python             	0x00000001065beae2 _PyEval_EvalFrameDefault + 6922
26  org.python.python             	0x00000001065c62cd _PyEval_EvalCodeWithName + 1698
27  org.python.python             	0x00000001065bcf35 PyEval_EvalCode + 51
28  org.python.python             	0x00000001065ba9bf builtin_exec + 563
29  org.python.python             	0x0000000106530ed0 _PyMethodDef_RawFastCallKeywords + 488
30  org.python.python             	0x000000010653045c _PyCFunction_FastCallKeywords + 41
31  org.python.python             	0x00000001065c5a6c call_function + 628
32  org.python.python             	0x00000001065beae2 _PyEval_EvalFrameDefault + 6922
33  org.python.python             	0x00000001065c62cd _PyEval_EvalCodeWithName + 1698
34  org.python.python             	0x0000000106530424 _PyFunction_FastCallKeywords + 212
35  org.python.python             	0x00000001065c5ad9 call_function + 737
36  org.python.python             	0x00000001065beae2 _PyEval_EvalFrameDefault + 6922
37  org.python.python             	0x00000001065c62cd _PyEval_EvalCodeWithName + 1698
38  org.python.python             	0x000000010653009c _PyFunction_FastCallDict + 444
39  org.python.python             	0x0000000106602cc0 pymain_run_module + 147
40  org.python.python             	0x0000000106601d2c pymain_main + 4091
41  org.python.python             	0x00000001066028e4 _Py_UnixMain + 56
42  libdyld.dylib                 	0x00007fff6ba83cc9 start + 1
```

**Comment by mfranberg** [comment]
This should be fixed by #25, if you still see a problem please open a new issue.
